src/Utilities/likelihood_informed.jl

```diff
@@ -0,0 +1,251 @@
+- `grad_type`[=:localsl]: how the gradient of the forward model at the samples will be approximated. Choose from `:linreg` (global linear regression) and `:localsl` (localized statistical linearization; see [Wacker, 2025]).
+- `use_data_as_samples`[=false]: if this parameter is `true`, then the data being processed (the training data for the emulator) will be used as the samples mentioned earlier. This means that they must be from the correct distribution corresponding to the chosen `alpha`. If this parameter is `false`, then the method expects `:samples_in` and `:samples_out` structure vectors that contain the samples instead.
+"""
+function likelihood_informed(; retain_KL, alpha = 0.0, grad_type = :localsl, use_data_as_samples = false)
+    if grad_type ∉ [:linreg, :localsl]
+        @error "Unknown grad_type=$grad_type"
+    end
+
+    LikelihoodInformed(nothing, nothing, nothing, (:retain_KL, retain_KL), alpha, grad_type, use_data_as_samples)
+end
+
+get_encoder_mat(li::LikelihoodInformed) = li.encoder_mat
+get_decoder_mat(li::LikelihoodInformed) = li.decoder_mat
+
+function initialize_processor!(
+    li::LikelihoodInformed,
+    in_data::MM,
+    out_data::MM,
+    ::Dict{Symbol, <:StructureMatrix},
+    output_structure_matrices::Dict{Symbol, <:StructureMatrix},
+    input_structure_vectors::Dict{Symbol, <:StructureVector},
+    output_structure_vectors::Dict{Symbol, <:StructureVector},
+    apply_to::AbstractString,
+) where {MM <: AbstractMatrix}
+    input_dim = size(in_data, 1)
+    output_dim = size(out_data, 1)
+
+    if isnothing(get_encoder_mat(li))
+        α = li.α
+        y = if α ≈ 0.0
+            # For α=0, it doesn't matter what this value is, so we avoid requiring its presence
+            zeros(size(out_data, 1))
+        else
+            get_structure_vec(output_structure_vectors, :observation)
+        end
+        samples_in, samples_out = if li.use_data_as_samples
+            (in_data, out_data)
+        else
+            (
+                get_structure_vec(input_structure_vectors, :samples_in),
+                get_structure_vec(output_structure_vectors, :samples_out),
+            )
+        end
+        obs_noise_cov = get_structure_mat(output_structure_matrices, :obs_noise_cov)
+        obs_whitened = if obs_noise_cov ≈ I
+            obs_noise_cov = I(output_dim)
+            true
+        else
+            @warn "Consider using decorrelate_structure_mat to gain obs_noise_cov = I before calling likelihood_informed"
+            false
+        end
+        noise_cov_inv = inv(obs_noise_cov)
+
+        li.apply_to = apply_to
+
+        grads = if li.grad_type == :linreg
+            grad = (samples_out .- mean(samples_out; dims = 2)) / (samples_in .- mean(samples_in; dims = 2))
+            fill(grad, size(samples_in, 2))
+        else
+            @assert li.grad_type == :localsl
```

Comment:
Could we have useful `throw(ArgumentError(...))`s for the user over asserts? In our experience it is much better to tell the user to use r to provide the correct args than to have asserts.